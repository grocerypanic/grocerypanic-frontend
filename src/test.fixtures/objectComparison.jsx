export const propCount = (hash, length) => {
  expect(Object.keys(hash).length).toBe(length);
};
