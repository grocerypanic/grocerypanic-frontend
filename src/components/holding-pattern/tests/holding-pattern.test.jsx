import React from "react";
import { render, cleanup } from "@testing-library/react";
import { propCount } from "../../../test.fixtures/objectComparison";

import HoldingPattern from "../holding-pattern.component";
import Spinner from "react-bootstrap/Spinner";

jest.mock("react-bootstrap/Spinner", () => ({
  __esModule: true,
  default: jest.fn(() => <div>MockSpinner</div>),
}));

let condition = false;

describe("Setup Environment", () => {
  let utils;
  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(cleanup);

  describe("When condition is false", () => {
    describe("with defaults", () => {
      beforeEach(() => {
        condition = false;
        utils = render(
          <HoldingPattern condition={condition}>
            Suppressed Component
          </HoldingPattern>
        );
      });

      it("should render with the correct message", () => {
        expect(utils.queryByText("Suppressed Component")).toBeTruthy();
        expect(utils.queryByTestId("HoldingPattern")).toBeFalsy();
        expect(Spinner).toBeCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });

    describe("with custom values", () => {
      beforeEach(() => {
        condition = false;
        utils = render(
          <HoldingPattern
            animation={"other1"}
            color={"other2"}
            condition={condition}
          >
            Suppressed Component
          </HoldingPattern>
        );
      });

      it("should render with the correct message", () => {
        expect(utils.queryByText("Suppressed Component")).toBeTruthy();
        expect(utils.queryByTestId("HoldingPattern")).toBeFalsy();
        expect(Spinner).toBeCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });
  });

  describe("When condition is true", () => {
    describe("with defaults", () => {
      beforeEach(() => {
        condition = true;
        utils = render(
          <HoldingPattern condition={condition}>
            Suppressed Component
          </HoldingPattern>
        );
      });

      it("should render the spinner instead", () => {
        expect(utils.queryByText("Suppressed Component")).toBeFalsy();
        expect(utils.queryByTestId("HoldingPattern")).toBeTruthy();
        expect(Spinner).toBeCalledTimes(1);
        const call = Spinner.mock.calls[0][0];
        propCount(call, 3);
        expect(call.animation).toBe("grow");
        expect(call.variant).toBe("success");
        expect(call.className).toBe("kindly-hang-in-there");
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });

    describe("with custom values", () => {
      beforeEach(() => {
        condition = true;
        utils = render(
          <HoldingPattern
            animation={"other1"}
            color={"other2"}
            condition={condition}
          >
            Suppressed Component
          </HoldingPattern>
        );
      });

      it("should render with the correct message", () => {
        expect(utils.queryByText("Suppressed Component")).toBeFalsy();
        expect(utils.queryByTestId("HoldingPattern")).toBeTruthy();
        expect(Spinner).toBeCalledTimes(1);
        const call = Spinner.mock.calls[0][0];
        propCount(call, 3);
        expect(call.animation).toBe("other1");
        expect(call.variant).toBe("other2");
        expect(call.className).toBe("kindly-hang-in-there");
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container.firstChild).toMatchSnapshot();
      });
    });
  });
});
