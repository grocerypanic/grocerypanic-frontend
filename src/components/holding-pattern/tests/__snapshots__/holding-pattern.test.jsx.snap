// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment When condition is false with custom values should match the snapshot on file (styles) 1`] = `Suppressed Component`;

exports[`Setup Environment When condition is false with defaults should match the snapshot on file (styles) 1`] = `Suppressed Component`;

exports[`Setup Environment When condition is true with custom values should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100vw;
  height: 80vh;
}

.c0 .kindly-hang-in-there {
  -webkit-transform: scale(3);
  -ms-transform: scale(3);
  transform: scale(3);
}

<ul
  class="c0"
  data-testid="HoldingPattern"
  height="80vh"
  scale="3"
>
  <div
    style="width: 75px; height: 75px;"
  >
    <div>
      MockSpinner
    </div>
  </div>
</ul>
`;

exports[`Setup Environment When condition is true with defaults should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100vw;
  height: 80vh;
}

.c0 .kindly-hang-in-there {
  -webkit-transform: scale(3);
  -ms-transform: scale(3);
  transform: scale(3);
}

<ul
  class="c0"
  data-testid="HoldingPattern"
  height="80vh"
  scale="3"
>
  <div
    style="width: 75px; height: 75px;"
  >
    <div>
      MockSpinner
    </div>
  </div>
</ul>
`;
